### DISCUSSION TITLE
How to seed using DHT?

### DISCUSSION THREAD

**Discussion by leoncwu** [anacrolix/torrent#827] [Q&A] [answered]
I am totally new to torrent. I want to share files among known machines.
1. build a DHT node (anacrolix/torrent/dht)
2. generate the torrent with the help of DHT
4. download files using the DHT
I have been reading through the examples but I still do not have a clear idea of what to do. Can I receive some help?

**Comment by anacrolix** [comment, marked as answer]
Your best bet would be to use https://github.com/anacrolix/torrent/blob/master/cmd/torrent/serve.go and https://github.com/anacrolix/torrent/blob/master/cmd/torrent/download.go, which are implementations for the cmd/torrent subcommands serve and download. They show examples of building a metainfo, and then downloading from an infohash. If you configure it correctly with DHT (the default, but you can force it by removing trackers), then the discovery part of the process will be through the DHT.

**Reply by leoncwu** [reply]
Thank you so much for your reply. I do not have to use a DHT table. A lot of people recommends openTracker but we have some license issue with it. Is there any way to establish a tracker node using github.com/anacrolix/torrent/tracker? I was also not able to find an example that works.

**Reply by anacrolix** [reply]
There are server components in there, but I don't stitch them together into a stand alone server implementation. I have such an implementation but it's closed source.

**Reply by leoncwu** [reply]
Just a follow-up question. I was not able to figure out how to configure with DHT. I noticed that the trackers are added this way in the specification:

`Trackers: [][]string{{`
`	trakcer1,`
`	tracker2,`
`}}, `

Is DHT node configured the same way? I am wondering if you could direct me to another example that uses a DHT when creating the torrent file.
Thank you in advance for your help.

**Reply by anacrolix** [reply]
No configuration for DHT is necessary when creating a torrent. Also the defaults will work fine for locating peers.

**Reply by leoncwu** [reply]
Thank you again for your prompt reply. I am new to BitTorren world. I have been trying to learn all the concepts. Somehow, I have the impression that DHT is a separate standalone node or service, and I need to give the BitTorrent server the address (URL) of the DHT node when creating the torrent file. Is this how DHT works in Anacrolix family? Could you explain a little bit more about how DHT works in this library? Where is this DHT stored after the torrent file is created? Thank you.

**Reply by anacrolix** [reply]
The DHT can operate independently of BitTorrent. It's a network that exists irrespective of any particular torrent. Operations in the DHT for BitTorrent are keyed by the infohash of the torrent. All the data is stored in the network, individual nodes contribute their resources to hosting data for arbitrary infohashes regardless of whether they are involved themselves with that infohash in BitTorrent.

**Reply by leoncwu** [reply]
Thank you. I am trying to get everything run first to understand this process. So, I did not modify the server and download example you shared in your first comment. I deleted only the part that provides the tracker addresses in server.go to force using the DHT. However, the download cannot start. Here are how to parameters are set up for the download.go part:
`{SaveMetainfos:true Mmap:true Seed:true Addr::8050 MaxUnverifiedBytes:1.0 MB UploadRate:100 MB DownloadRate:100 MB PackedBlocklist: PublicIP:<nil> Progress:true PieceStates:true Quiet:true Stats:true Dht:true PortForward:true TcpPeers:true UtpPeers:true Webtorrent:true DisableWebseeds:true RequireFastExtension:true Ipv4:true Ipv6:true Pex:true LinearDiscard:false TestPeer:[] File:[] Torrent:[magnet:?xt=urn:btih:f39d9117ba0ca73d50f47430d2532030a827ad73&dn=test.txt]}`
The download stuck when getting the torrent info.
`297ns: getting torrent info for "test.txt"`
Does any parameter setup cause the issue? Thank you again for your help.

**Reply by anacrolix** [reply]
You will need seeders available on the network for you torrent, that your client can connect to. I can see that you are creating your own torrent, connectivity to a single other peer can be dubious. Try with a very popular torrent, disable trackers so that DHT will be the only source of peers and see how it goes. After you get that working, then work on swarms with few or singular peers.
